util/metrics/metrics.go

```diff
@@ -2,59 +2,71 @@ package metrics
 
 import "github.com/prometheus/client_golang/prometheus"
 
+// This is helpful for tests to ensure there are never uninitialized values.
+func init() {
+	RegisterPrometheusMetrics("uninitialized")
+}
+
 // RegisterPrometheusMetrics register all prometheus metrics with the global
 // metrics handler.
-func RegisterPrometheusMetrics() {
-	for _, c := range collectors {
-		prometheus.Register(c)
+func RegisterPrometheusMetrics(subsystem string) {
+	deregister()
```

Comment:
why is deregister() needed?

Reply:
It's mainly for tests, which need to make sure these objects are instantiated even if the main init functions are not called. I'll update with a comment.
```suggestion
	// deregister metric objects in case the register function was called more than once.
	// This helps with testing.
	deregister()
```